service/controller/v1/resource/chartconfig/desired.go

```diff
@@ -41,30 +50,31 @@ func (r *Resource) newChartCR(ctx context.Context, customResource v1alpha1.Relea
 			APIVersion: "core.giantswarm.io",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: key.OperatorChartName(customResource),
+			Name: fmt.Sprintf("%s-chart", authority.Name),
 			Labels: map[string]string{
-				key.LabelApp:          key.OperatorName(customResource),
-				key.LabelManagedBy:    key.ProjectName,
-				key.LabelOrganization: key.OrganizationName,
-				key.LabelServiceType:  key.ServiceTypeManaged,
+				key.LabelApp:            authority.Name,
+				key.LabelManagedBy:      key.ProjectName,
+				key.LabelOrganization:   key.OrganizationName,
+				key.LabelReleaseVersion: key.ReleaseVersion(customResource),
+				key.LabelServiceType:    key.ServiceTypeManaged,
 			},
 		},
 		Spec: v1alpha1.ChartConfigSpec{
 			Chart: v1alpha1.ChartConfigSpecChart{
-				Channel: key.OperatorChannelName(customResource),
+				Channel: fmt.Sprintf("%s-%s", authority.Name, strings.Replace(authority.Version, ".", "-", -1)),
 				ConfigMap: v1alpha1.ChartConfigSpecConfigMap{
 					Name:            c.ConfigMap.Name,
 					Namespace:       c.ConfigMap.Namespace,
 					ResourceVersion: c.ConfigMap.ResourceVersion,
 				},
-				Name:      key.OperatorChartName(customResource),
+				Name:      fmt.Sprintf("%s-chart", authority.Name),
 				Namespace: metav1.NamespaceSystem,
 				Secret: v1alpha1.ChartConfigSpecSecret{
 					Name:            c.Secret.Name,
 					Namespace:       c.Secret.Namespace,
 					ResourceVersion: c.Secret.ResourceVersion,
 				},
-				Release: key.ReleaseName(customResource),
+				Release: fmt.Sprintf("%s-%s", authority.Name, strings.Replace(authority.Version, ".", "-", -1)),
```

Comment:
I figured that the `Channel` and the `Release` are actually the same in our case here. Can anybody confirm?

Reply:
Yes this should be the case because we only expect to run one release per channel.

Reply:
This fmt stuff should also be inside the key package IMO.

Reply:
I will go with this and provide PRs to use key functions from `apiextensions`.

Reply:
Not into the details but that makes sense to me. 

Stating the obvious but release is used as the Helm release name. One gotcha we hit was you can't have multiple releases with the same name. Even if they are installed in different namespaces.

Reply:
See https://github.com/giantswarm/apiextensions/pull/143.